forem/forem
# Heavy Confirmation Flow for Broadcast-Destroy

## What type of PR is this? (check all applicable)


## Description
In this PR, I refactor the process of deleting Broadcasts, by presenting a Confirmation Modal before completing the destructive action. This flow is called "Heavy Confirmation" in the [Save-Confirmation RFC](https://github.com/forem/rfcs/blob/237480c4b915a64fc17df8f525b0a429185a1605/text/0078-admin-save-confirmation.md).
I also generalize the `alert_controller.js` file so that other custom alerts (eg success) can be created as needed.

## Related Issues and PRs
- https://github.com/forem/forem/pull/14684

## QA Instructions, Screenshots, Recordings

1. Login to `http://localhost:xxxx/admin/` as a `super_admin`. Navigate to `Advanced`, then `Broadcasts`. Choose a broadcast.
2. Click the `Destroy` button. A modal titled `Confirm changes` should popup.
3. Click either the `Discard Changes` or `X` button at the top right corner of the modal. The modal should close without any actions being taken.
4. Click any `Destroy` button again. Type in some random text, then click `Confirm changes`. A yellow warning notice should appear, telling you that your text does not match the expected entry.
5. Now type in the correct confirmation text and click `Confirm Changes`. The expected behavior is that the page should redirect to the general Broadcasts page `(/admin/advanced/broadcasts)` and a green success alert should appear at the top.

### UI accessibility concerns?
All UI accessibility concerns will be addressed through PR review.

## Added/updated tests?

      have not been included_


## [Forem core team only] How will this change be communicated?

      or in a [forem.dev](http://forem.dev) post


## What gif best describes this PR or how it makes you feel?
![Hey Arnold clip](https://media.giphy.com/media/xT1R9TJzgptMjU3Asw/giphy.gif)
